jdk/src/windows/back/linker_md.c

```diff
@@ -51,11 +57,14 @@ static void dll_build_name(char* buffer, size_t buflen,
     path = strtok_s(paths_copy, PATH_SEPARATOR, &next_token);
 
     while (path != NULL) {
-        _snprintf(buffer, buflen, "%s\\%s.dll", path, fname);
-        if (_access(buffer, 0) == 0) {
-            break;
+        /* If the path will cause a buffer overflow skip it. This path is invalid. */
+        if (path + (int)strlen(fname) + 10 <= buflen) {
```

Comment:
This needs to use the length of the path. 

Could pull strlen(fname) out of the loop.

Not sure of the purpose of +10. I see +6 for the extra chars added plus the null terminator.

Reply:
`strlen` returns `size_t`: the cast to `int` is inappropriate and should not be necessary.